### ISSUE TITLE
GetSymbolInfo for ValueTuple<int,int> says the type is from source

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#14524]
When I reference the assembly that defines ValueTuple, and then instantiate it in source, the SemanticModel says that the type is defined in source (i.e. all of its locations are "from source").

/cc @dpoeschl @jmarolf @CyrusNajmabadi @VSadov

**Comment by dpoeschl** [comment]
@gafter On second validation, it looks like there are no locations in this case, which surprises Rename into crashing since it was invoked on a token. This is probably the desired behavior?

**Comment by gafter** [comment]
@dpoeschl Why would you expect that to be a location for the symbol? Did the semantic model report that as a declaration point for the type (i.e. GetDeclaredSymbol returning something)?

**Comment by dpoeschl** [comment]
No, GetDeclaredSymbol didn't return anything. It shouldn't be a Location of the symbol in this case. 

What I meant to say was: For symbols that have no Locations in metadata, it expected the token to be a Location of the symbol returned by GetSymbolInfo when called on that token. This is no longer the case. Maybe it was a faulty assumption all along and there are other examples where this falls down, we've just never hit them...

**Comment by jaredpar** [comment]
What is the impact of this bug in the current state of the product?  Need to understand if it still meets the RTM bar.

**Comment by gafter** [comment]
This would likely be fixed as a side-effect of the changes proposed for https://github.com/dotnet/roslyn/issues/20648
